### ISSUE TITLE
Q-Line Similar Unreal, Bahamut, PTLink

### ISSUE THREAD

**Issue by TuSuNaMi** [inspircd/inspircd#756]
Hello;

I need a Q-Line 'd operator nicks. After the ircop enter the oper he/she must be use a Q-Lined nick. If possible this version please send sample me.

Thanks

**Comment by sebastiannielsen** [comment]
Here you have:
https://github.com/inspircd/inspircd-extras/blob/master/2.0/m_badnicks.cpp

Just use that module, and specify the oper nicks in "badnicks=" option for the connect class belongning to "normal users".
For the connect class belongning to your opers, just leave "badnicks=" out.
Do NOT write oper nicks in the <badnick> server configuration option.


Robby-: The problem is that once you connect the server cannot know you are an oper. Thats why you have to use my module m_badnicks.cpp that uses IP-numbers to identify exceptions.

**Comment by SISheogorath** [comment]
I wonder what you try to prevent. Probably fooling of other users. The solution were always and still are: services. They have all tools to prevent "nick-stealing".

I don't see why we need to handle this in the IRCd. From the IRCd perspective it's completely valid for a user to use the same nick as an oper. Because what makes people an oper or not is not the nick, but the authentication and that's indicated to users by the whois.

When we come to the point where we integrate services with the IRCd, I think this would maybe come in handy, but we are not about to do this within the next few releases.

**Comment by Robby-** [comment]
@sebastiannielsen: Yeah, its an inconvenience, but not really an issue if you just keep that in mind. Just /OPER, and then change your nickname to a Q-lined nick.

@SISheogorath: I still like the idea of a new QLINE override option as I suggested 4 years ago. That should still be very much within scope for implementing in inspircd.

**Comment by sebastiannielsen** [comment]
Robby-: What do you think about the solution I gave, by using the module m_badnicks.so? I think the solution is almost identical, but the oper don't need to change the nick but can join directly with the desired nick.

**Comment by Robby-** [comment]
@sebastiannielsen: Its a good solution, one that provides some more flexibility. But for some people it may be more than what they want or need, so a new QLINE override option may be enough for them. These 2 things can co-exist just fine.
